File: address-geocode-international-dot-net-examples/ReverseSearchRestSdkExample.cs
using address_geocode_international_dot_net.REST;

namespace address_geocode_international_dot_net_examples
{
    internal static class ReverseSearchRestSdkExample
    {
        public static void Go(string licenseKey, bool isLive)
        {
            Console.WriteLine("\r\n--------------------------------------------------------");
            Console.WriteLine("Address Geocode International - ReverseSearch - REST SDK");
            Console.WriteLine("--------------------------------------------------------");

            ReverseSearchClient.ReverseSearchInput reverseSearchInput = new(
                Latitude: "40.748369",
                Longitude: "-73.984853",
                SearchRadius: "5",
                Country: "US",
                MaxResults: "2",
                SearchType: "All",
                LicenseKey: licenseKey,
                IsLive: isLive,
                TimeoutSeconds: 15
            );

            Console.WriteLine("\r\n* Input *\r\n");
            Console.WriteLine($"Latitude    : {reverseSearchInput.Latitude}");
            Console.WriteLine($"Longitude   : {reverseSearchInput.Longitude}");
            Console.WriteLine($"SearchRadius: {reverseSearchInput.SearchRadius}");
            Console.WriteLine($"Country     : {reverseSearchInput.Country}");
            Console.WriteLine($"MaxResults  : {reverseSearchInput.MaxResults}");
            Console.WriteLine($"Search Type : {reverseSearchInput.SearchType}");
            Console.WriteLine($"Is Live     : {reverseSearchInput.IsLive}");
            Console.WriteLine($"License Key : {reverseSearchInput.LicenseKey}");
            Console.WriteLine($"Timeout     : {reverseSearchInput.TimeoutSeconds}");

            AGIReverseSearchResponse response = ReverseSearchClient.Invoke(reverseSearchInput);

            if (response.Error is null)
            {
                Console.WriteLine("\r\n* Results *\r\n");

                if (response.SearchInfo != null)
                {
                    Console.WriteLine("Search Info:");
                    Console.WriteLine($"Status             : {response.SearchInfo.Status}");
                    Console.WriteLine($"Number Of Locations: {response.SearchInfo.NumberOfLocations}");
                    Console.WriteLine($"Notes              : {response.SearchInfo.Notes}");
                    Console.WriteLine($"Notes Desc         : {response.SearchInfo.NotesDesc}");
                    Console.WriteLine($"Warnings           : {response.SearchInfo.Warnings}");
                    Console.WriteLine($"WarningDesc        : {response.SearchInfo.WarningDesc}");
                    Console.WriteLine();
                }

                if (response.Locations != null && response.Locations.Count > 0)
                {
                    int index = 1;
                    foreach (var location in response.Locations)
                    {
                        Console.WriteLine($"\r\nLocation #{index++}\r\n");
                        Console.WriteLine($"\tPrecision Level                : {location.PrecisionLevel}");
                        Console.WriteLine($"\tType                           : {location.Type}");
                        Console.WriteLine($"\tLatitude                       : {location.Latitude}");
                        Console.WriteLine($"\tLongitude                      : {location.Longitude}");

                        if (location.AddressComponents != null)
                        {
                            Console.WriteLine($"\tPremiseNumber                  : {location.AddressComponents.PremiseNumber}");
                            Console.WriteLine($"\tThoroughfare                   : {location.AddressComponents.Thoroughfare}");
                            Console.WriteLine($"\tDoubleDependentLocality        : {location.AddressComponents.DoubleDependentLocality}");
                            Console.WriteLine($"\tDependentLocality              : {location.AddressComponents.DependentLocality}");
                            Console.WriteLine($"\tLocality                       : {location.AddressComponents.Locality}");
                            Console.WriteLine($"\tAdministrativeArea1            : {location.AddressComponents.AdministrativeArea1}");
                            Console.WriteLine($"\tAdministrativeArea1Abbreviation: {location.AddressComponents.AdministrativeArea1Abbreviation}");
                            Console.WriteLine($"\tAdministrativeArea2            : {location.AddressComponents.AdministrativeArea2}");
                            Console.WriteLine($"\tAdministrativeArea2Abbreviation: {location.AddressComponents.AdministrativeArea2Abbreviation}");
                            Console.WriteLine($"\tAdministrativeArea3            : {location.AddressComponents.AdministrativeArea3}");
                            Console.WriteLine($"\tAdministrativeArea3Abbreviation: {location.AddressComponents.AdministrativeArea3Abbreviation}");
                            Console.WriteLine($"\tAdministrativeArea4            : {location.AddressComponents.AdministrativeArea4}");
                            Console.WriteLine($"\tAdministrativeArea4Abbreviation: {location.AddressComponents.AdministrativeArea4Abbreviation}");
                            Console.WriteLine($"\tPostalCode                     : {location.AddressComponents.PostalCode}");
                            Console.WriteLine($"\tCountry                        : {location.AddressComponents.Country}");
                            Console.WriteLine($"\tCountryISO2                    : {location.AddressComponents.CountryISO2}");
                            Console.WriteLine($"\tCountryISO3                    : {location.AddressComponents.CountryISO3}");
                            Console.WriteLine($"\tPlaceName                      : {location.AddressComponents.PlaceName}");
                            Console.WriteLine($"\tGoogleMapsURL                  : {location.AddressComponents.GoogleMapsURL}");
                            Console.WriteLine($"\tIsUnincorporated               : {location.AddressComponents.IsUnincorporated}");
                            Console.WriteLine($"\tStateFIPS                      : {location.AddressComponents.StateFIPS}");
                            Console.WriteLine($"\tCountyFIPS                     : {location.AddressComponents.CountyFIPS}");
                            Console.WriteLine($"\tCensusTract                    : {location.AddressComponents.CensusTract}");
                            Console.WriteLine($"\tCensusBlock                    : {location.AddressComponents.CensusBlock}");
                            Console.WriteLine($"\tCensusGeoID                    : {location.AddressComponents.CensusGeoID}");
                            Console.WriteLine($"\tClassFP                        : {location.AddressComponents.ClassFP}");
                            Console.WriteLine($"\tCongressCode                   : {location.AddressComponents.CongressCode}");
                            Console.WriteLine($"\tSLDUST                         : {location.AddressComponents.SLDUST}");
                            Console.WriteLine($"\tSLDLST                         : {location.AddressComponents.SLDLST}");
                            Console.WriteLine($"\tTimezone_UTC                   : {location.AddressComponents.TimeZone_UTC}");
                        }
                    }
                }
                else
                {
                    Console.WriteLine("No locations found.");
                }
            }
            else
            {
                Console.WriteLine("\r\n* Error *\r\n");

                Console.WriteLine($"Error Type     : {response.Error.Type}");
                Console.WriteLine($"Error Type Code: {response.Error.TypeCode}");
                Console.WriteLine($"Error Desc     : {response.Error.Desc}");
                Console.WriteLine($"Error Desc Code: {response.Error.DescCode}");
            }
        }
    }
}

File: address-geocode-international-nodejs-examples/reverse_search_rest_sdk_example.js
import { ReverseSearchClient } from '../address-geocode-international/REST/reverse_search_rest.js';

async function ReverseSearchRestGo(licenseKey, isLive) {
    console.log("\n--------------------------------------------------------");
    console.log("Address Geocode International - ReverseSearch - REST SDK");
    console.log("--------------------------------------------------------");

    const latitude = "34.02984234473112";
    const longitude = "-118.26876271804397"; 
    const searchRadius = "";
    const country = "";
    const maxResults = "";
    const searchType = "BestMatch";
    const timeoutSeconds = 15;

    console.log("\n* Input *\n");
    console.log(`Latitude         : ${latitude}`);
    console.log(`Longitude        : ${longitude}`);
    console.log(`SearchRadius     : ${searchRadius}`);
    console.log(`Country          : ${country}`);
    console.log(`MaxResults       : ${maxResults}`);
    console.log(`SearchType       : ${searchType}`);
    console.log(`License Key      : ${licenseKey}`);
    console.log(`Is Live          : ${isLive}`);
    console.log(`Timeout Seconds  : ${timeoutSeconds}`);

    try {
        const response = await ReverseSearchClient.invoke(
            latitude,
            longitude,
            searchRadius,
            country,
            maxResults,
            searchType,
            licenseKey,
            isLive,
            timeoutSeconds
        );

        if (response.Error) {
            console.log("\n* Error *\n");
            console.log(`Error Type    : ${response.Error.Type}`);
            console.log(`Error TypeCode: ${response.Error.TypeCode}`);
            console.log(`Error Desc    : ${response.Error.Desc}`);
            console.log(`Error DescCode: ${response.Error.DescCode}`);
            return;
        }

        console.log("\n* Search Info *\n");
        if (response && response.SearchInfo) {
            console.log(`Status           : ${response.SearchInfo.Status}`);
            console.log(`NumberOfLocations: ${response.SearchInfo.NumberOfLocations}`);
            console.log(`Notes            : ${response.SearchInfo.Notes}`);
            console.log(`NotesDesc        : ${response.SearchInfo.NotesDesc}`);
            console.log(`Warnings         : ${response.SearchInfo.Warnings}`);
            console.log(`WarningDesc      : ${response.SearchInfo.WarningDesc}`);
        } else {
            console.log("No search info found.");
        }

        console.log("\n* Locations *\n");
        if (response && response.Locations && response.Locations.length > 0) {
            response.Locations.forEach((location) => {
                console.log(`PrecisionLevel   : ${location.PrecisionLevel}`);
                console.log(`Type             : ${location.Type}`);
                console.log(`Latitude         : ${location.Latitude}`);
                console.log(`Longitude        : ${location.Longitude}`);

                console.log("\n* Address Components *");
                if (location.AddressComponents) {
                    console.log(`PremiseNumber           : ${location.AddressComponents.PremiseNumber}`);
                    console.log(`Thoroughfare            : ${location.AddressComponents.Thoroughfare}`);
                    console.log(`DoubleDependentLocality : ${location.AddressComponents.DoubleDependentLocality}`);
                    console.log(`DependentLocality       : ${location.AddressComponents.DependentLocality}`);
                    console.log(`Locality                : ${location.AddressComponents.Locality}`);
                    console.log(`AdministrativeArea1     : ${location.AddressComponents.AdministrativeArea1}`);
                    console.log(`AdministrativeArea1Abbr : ${location.AddressComponents.AdministrativeArea1Abbreviation}`);
                    console.log(`AdministrativeArea2     : ${location.AddressComponents.AdministrativeArea2}`);
                    console.log(`AdministrativeArea2Abbr : ${location.AddressComponents.AdministrativeArea2Abbreviation}`);
                    console.log(`AdministrativeArea3     : ${location.AddressComponents.AdministrativeArea3}`);
                    console.log(`AdministrativeArea3Abbr : ${location.AddressComponents.AdministrativeArea3Abbreviation}`);
                    console.log(`AdministrativeArea4     : ${location.AddressComponents.AdministrativeArea4}`);
                    console.log(`AdministrativeArea4Abbr : ${location.AddressComponents.AdministrativeArea4Abbreviation}`);
                    console.log(`PostalCode              : ${location.AddressComponents.PostalCode}`);
                    console.log(`Country                 : ${location.AddressComponents.Country}`);
                    console.log(`CountryISO2             : ${location.AddressComponents.CountryISO2}`);
                    console.log(`CountryISO3             : ${location.AddressComponents.CountryISO3}`);
                    console.log(`GoogleMapsURL           : ${location.AddressComponents.GoogleMapsURL}`);
                    console.log(`PlaceName               : ${location.AddressComponents.PlaceName}`);
                    console.log(`IsUnincorporated        : ${location.AddressComponents.IsUnincorporated}`);
                    console.log(`StateFIPS               : ${location.AddressComponents.StateFIPS}`);
                    console.log(`CountyFIPS              : ${location.AddressComponents.CountyFIPS}`);
                    console.log(`CensusTract             : ${location.AddressComponents.CensusTract}`);
                    console.log(`CensusBlock             : ${location.AddressComponents.CensusBlock}`);
                    console.log(`CensusGeoID             : ${location.AddressComponents.CensusGeoID}`);
                    console.log(`ClassFP                 : ${location.AddressComponents.ClassFP}`);
                    console.log(`CongressCode            : ${location.AddressComponents.CongressCode}`);
                    console.log(`SLDUST                  : ${location.AddressComponents.SLDUST}`);
                    console.log(`SLDLST                  : ${location.AddressComponents.SLDLST}`);
                    console.log(`Timezone_UTC            : ${location.AddressComponents.Timezone_UTC}`);
                } else {
                    console.log("No address components found.");
                }
            });
        } else {
            console.log("No locations found.");
        }
    } catch (e) {
        console.log("\n* Error *\n");
        console.log(`Error Message: ${e.message}`);
    }
}

export { ReverseSearchRestGo };